### PR TITLE
fix issues of naming of some methods

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Description

Previous implement has this scenario:
A post operation with operationId `Quotas_Update` returns a collection, the generated code will have a function corresponds this operation with the name of `GetQuotaate` - which is a truncate of the word `Update`. This PR fixes this.

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs [hunk 1]
```diff
@@ -288,5 +289,18 @@ public static string FirstCharToLowerCase(this string str)
...
+
+        public static string FirstCharToUpperCase(this string str)
+        {
+            if (string.IsNullOrEmpty(str) || char.IsUpper(str[0]))
+                return str;
+
+            return char.ToUpper(str[0]) + str.Substring(1);
+        }
+
+        public static IEnumerable<string> SplitByCamelCase(this string camelCase)
```
Comment (archerzz) [inline src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs:301]: Shall we check `null` here?

Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs:301]: Actually we are doing things in explicit null now - therefore the input value can never be null.

Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs:301]: And we do not need to check whether this is empty here, I suppose it is reasonable that this returns empty collection if we get empty input.

File: test/TestProjects/MgmtListMethods/Generated/Configuration.json [hunk 2]
```diff
@@ -14,16 +14,22 @@
     "OneResources": "Microsoft.Fake/oneResources",
     "SecondResources": "Microsoft.Fake/fakes/secondResources",
     "SubFakes": "Microsoft.Fake/fakes/subFakes",
-    "SharedGalleries": "Microsoft.Compute/locations/sharedGalleries"
+    "SharedGalleries": "Microsoft.Compute/locations/sharedGalleries",
+    "Quotas": "Microsoft.Fake/locations/updateQuotas",
```
Comment (allenjzhang) [inline test/TestProjects/MgmtListMethods/Generated/Configuration.json:18]: nit: should be locations/quotas.

File: test/TestProjects/MgmtParent/readme.md [hunk 3]
```diff
@@ -13,8 +13,6 @@ head-as-boolean: false
 payload-flattening-threshold: 2
 operation-group-to-resource-type:
    VirtualMachineExtensionImages: Microsoft.Compute/locations/publishers/vmextension
-operation-group-to-resource:
```
Comment (allenjzhang) [inline test/TestProjects/MgmtParent/readme.md:16]: Is this related to the naming fix?

File: test/TestProjects/ResourceRename/Generated/Extensions/ResourceGroupExtensions.cs [hunk 4]
```diff
@@ -67,7 +67,7 @@ public static async Task<Response<SshPublicKeyInfo>> GetSshPublicKeyateAsync(thi
         /// <param name="keyData"> SSH public key certificate used to authenticate with the VM through ssh. The key needs to be at least 2048-bit and in ssh-rsa format. &lt;br&gt;&lt;br&gt; For creating ssh keys, see [Create SSH keys on Linux and Mac for Linux VMs in Azure](https://docs.microsoft.com/azure/virtual-machines/virtual-machines-linux-mac-create-ssh-keys?toc=%2fazure%2fvirtual-machines%2flinux%2ftoc.json). </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="sshPublicKeyName"/> is null. </exception>
-        public static Response<SshPublicKeyInfo> GetSshPublicKeyate(this ResourceGroupOperations resourceGroup, string sshPublicKeyName, string path = null, string keyData = null, CancellationToken cancellationToken = default)
+        public static Response<SshPublicKeyInfo> CreateSshPublicKey(this ResourceGroupOperations resourceGroup, string sshPublicKeyName, string path = null, string keyData = null, CancellationToken cancellationToken = default)
```
Comment (m-nash) [inline test/TestProjects/ResourceRename/Generated/Extensions/ResourceGroupExtensions.cs:70]: Why are these methods here if?  It looks like SshPublicKey is now a resource?  If so shouldn't it have its own container / operations to host the methods for it?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: test/TestProjects/ResourceRename/Generated/Extensions/ResourceGroupExtensions.cs (hunk 4 above)
Comment (ArcturusZhang) [reply test/TestProjects/ResourceRename/Generated/Extensions/ResourceGroupExtensions.cs:70]: yes, but this test project is not for this - I am unsure why we marked this NonResource in this test project. Do you have context?

Comment (allenjzhang) [reply test/TestProjects/ResourceRename/Generated/Extensions/ResourceGroupExtensions.cs:70]: The test project jsons are only used for validating specific scenarios. The annotation may/not be there for reason. Let's not worry about the directive correctness at this point.

**Review by m-nash — APPROVED** [review]
